src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3658,7 +3662,7 @@ void G1CollectedHeap::pre_evacuate_collection_set(G1EvacuationInfo& evacuation_i
   _bytes_used_during_gc = 0;
 
   _expand_heap_after_alloc_failure = true;
-  _evacuation_failed = false;
+  Atomic::store(&_num_regions_failed_evacuation, 0u);
```

Comment:
This could just be `_num_regions_failed_evacuation = 0;` right? Or am I missing something? It sticks out a bit and making one wonder if there is someone else change this value as well.